cli/src/main/java/dev/starfix/Starfix.java

```diff
@@ -402,17 +406,21 @@ public  void launch_editor(Path directory, String ide, String path, String fileP
 
         public  void launch_editor(Path directory, String ide, String path, String filePath) throws IOException, InterruptedException {
             if(filePath.indexOf("#")>0){
-                filePath = filePath.replace("#L","#");
+                filePath = filePath.replace("#L",":");
                 // eclipse.exe file.txt:22
-                if(ide.equals("eclipse")){
-                    filePath = filePath.replace("#",":");
-                    runCommand(directory.getParent(), ide,path,filePath);
-                }
-
             }
-            else{
-                runCommand(directory.getParent(),ide,path,filePath);
+            runCommand(directory.getParent(), ide,path,filePath);
+        }
+    }
+
+    public static class CustomIDE extends IDE{
+
+        public  void launch_editor(Path directory, String ide, String path, String filePath) throws IOException, InterruptedException {
+            if(filePath.indexOf("#")>0){
```

Comment:
why these eclipse related checks?

Reply:
That's not an eclipse related check . Its intended for URL having line numbers at the end. 

Example: https://github.com/jbangdev/jbang/blob/main/build.gradle#L12

The filePath we get will be:`jbang/blob/main/build.gradle#L12`  

Using this check we will convert it something like:
`jbang/blob/main/build.gradle:12`

I thought it'd be more meaningful (we don't know which IDE is being used but apparently this format looked more meaningful for IDEs to support) .

Reply:
okey - really should let each IDE subclass decide how they want to treat this.

Reply:
but i guess for custom IDE it can make sense.